base_layer/core/src/blocks/block_header.rs

```diff
@@ -382,6 +384,48 @@ pub(crate) mod hash_serializer {
     }
 }
 
+impl ConsensusEncoding for BlockHeader {
+    fn consensus_encode<W: Write>(&self, writer: &mut W) -> Result<usize, io::Error> {
+        let mut written = self.version.consensus_encode(writer)?;
+        written += self.height.consensus_encode(writer)?;
+        written += self.prev_hash.consensus_encode(writer)?;
+        written += self.timestamp.as_u64().consensus_encode(writer)?;
+        written += self.output_mr.consensus_encode(writer)?;
```

Comment:
For all hashes, we need to use fixed size encoding. Unfortunately, since we use a dynamic vec for hashes, this is unwieldy. The following code (from `BlockHeader::merge_mining_hash`) will do it:

```rust
            // TODO: Cleanup if/when we migrate to fixed 32-byte array type for hashes
            copy_into_fixed_array::<_, 32>(&self.output_mr)
                .unwrap()
                .consensus_encode(&mut hasher)
                .unwrap();
```

For decoding:

```rust
let output_mr = <[u8;32] as ConsensusDecoding>::consensus_decode(reader)?;
header.output_mr = output_mr.to_vec();
```

Reply:
Plan on moving to fixed hashes at some point